src/adaptix/_internal/morphing/facade/provider.py

```diff
@@ -439,3 +441,35 @@ def default_dict(pred: Pred, default_factory: Callable) -> Provider:
     :param default_factory: default_factory parameter of the ``defaultdict`` instance to be created by the loader
     """
     return bound(pred, DefaultDictProvider(default_factory))
+
+
+def datetime_by_timestamp(pred: Pred, tz: Optional[timezone] = timezone.utc) -> Provider:
+    """Provider that can load/dump datetime object from/to UNIX timestamp.
+
+    :param pred: Predicate specifying where the provider should be used.
+        See :ref:`predicate-system` for details.
+    :param tz: tz parameter which will be passed to the datetime.fromtimestamp method.
+    """
+
+    return bound(pred, DatetimeTimestampProvider(tz))
+
+
+def datetime_by_format(pred: Pred, fmt: str) -> Provider:
+    """Provider that can load/dump datetime object from/to format string e.g "%d/%m/%y %H:%M"
+
+    :param pred: Predicate specifying where the provider should be used.
+        See :ref:`predicate-system` for details.
+    :param fmt: format parameter which will be passed to datetime.strptime method.
+    """
+    return bound(pred, DatetimeFormatProvider(fmt))
+
+
+def date_by_timestamp(pred: Pred) -> Provider:
+    """Provider that can load date object from UNIX timestamp.
+    Note that date objects can`t be dumped to the UNIX timestamp
+
+    :param pred: Predicate specifying where the provider should be used.
+        See :ref:`predicate-system` for details.
+    """
+    return bound(pred, DateTimestampProvider())
```

Comment:
I think that a lot of people want to apply this rule to all datetimes or dates, so we need zero arguments signature.
And second argument will become keyword-only.

Reply:
To create zero arguments signature, pred argument have to default value `P.ANY`